src/legacy/core_plugins/kbn_vislib_vis_types/public/editors/point_series.tsx

```diff
@@ -77,9 +79,61 @@ function PointSeriesOptions(props: VisOptionsProps<BasicVislibParams>) {
             }
           />
         )}
+
+        <SwitchOption
+          label={i18n.translate('kbnVislibVisTypes.editors.pointSeries.thresholdLine.show', {
+            defaultMessage: 'Show threshold line',
+          })}
+          paramName="show"
+          value={stateParams.thresholdLine.show}
+          setValue={(paramName, value) =>
+            setValue('thresholdLine', { ...stateParams.thresholdLine, [paramName]: value })
+          }
+        />
+
+        {stateParams.thresholdLine.show && (
+          <NumberInputOption
+            label={i18n.translate('kbnVislibVisTypes.editors.pointSeries.thresholdLine.valueLabel', {
+              defaultMessage: 'Threshold value',
+            })}
+            paramName="value"
+            value={stateParams.thresholdLine.value}
+            setValue={(paramName, value) =>
+              setValue('thresholdLine', { ...stateParams.thresholdLine, [paramName]: value })
+            }
+          />
+        ) && (
```

Comment:
Nitpick: We usually don't use that pattern for showing multiple options. Could we instead wrap all of them in a `React.Fragment` instead and only have the one `&&` on top?